distributed/scheduler.py

```diff
@@ -2212,18 +2212,17 @@ def decide_worker_non_rootish(self, ts: TaskState) -> WorkerState | None:
             wp_vals = cast("Sequence[WorkerState]", worker_pool.values())
             n_workers: int = len(wp_vals)
             if n_workers < 20:  # smart but linear in small case
-                ws = min(wp_vals, key=operator.attrgetter("occupancy"))
+                ws = min(wp_vals, key=self.worker_objective_no_deps)
                 assert ws
-                if ws.occupancy == 0:
+                if sum(self.worker_objective_no_deps(ws)) == 0:
                     # special case to use round-robin; linear search
-                    # for next worker with zero occupancy (or just
-                    # land back where we started).
+                    # for next empty worker (or just land back where we started).
                     wp_i: WorkerState
                     start: int = self.n_tasks % n_workers
                     i: int
                     for i in range(n_workers):
                         wp_i = wp_vals[(i + start) % n_workers]
-                        if wp_i.occupancy == 0:
+                        if sum(self.worker_objective_no_deps(wp_i)) == 0:
```

Comment:
I'm keeping this round-robin behavior around for now, but hopefully we'd remove it in a separate PR later.